src/backend/utils.ts

```diff
@@ -1306,19 +1306,76 @@ export const writeConfig = (appName: string, config: Partial<AppSettings>) => {
   }
 }
 
+// Helper function to check permissions
+async function checkPermissions(src: string, dest: string) {
+  try {
+    // Check if we can read from source
+    await access(src, constants.R_OK)
+
+    // Check if destination parent directory exists and is writable
+    const destDir = dirname(dest)
+    try {
+      await access(destDir, constants.W_OK)
+    } catch {
+      throw new Error(
+        `No write permission for destination directory: ${destDir}`
+      )
+    }
+  } catch (error) {
+    if (error instanceof Error) {
```

Comment:
I don't think we need this check, error will always be an instances of the Error class by default

Reply:
This is necessary to get the error type properly, without it we dont have access to `error.message`.
But I changed it since in this case would be good to get the whole Error object.